src/data/dfinsupp.lean

```diff
@@ -408,6 +416,38 @@ have h4 : f + single i b = single i b + f,
   { simp [H] } },
 eq.rec_on h4 $ ha i b f h1 h2 h3
 
+@[simp] lemma add_closure_Union_range_single :
+  add_submonoid.closure (⋃ i : ι, set.range (single i : β i → (Π₀ i, β i))) = ⊤ :=
+top_unique $ λ x hx, (begin
+  apply dfinsupp.induction x,
+  exact add_submonoid.zero_mem _,
+  exact λ a b f ha hb hf, add_submonoid.add_mem _
+    (add_submonoid.subset_closure $ set.mem_Union.2 ⟨a, set.mem_range_self _⟩) hf
+end)
```

Comment:
This worked in term mode for `finsupp`, but the elaborator couldn't fill out the underscores unless I did it in tactic mode.